packages/manager/apps/pci-private-network/src/api/hooks/useNetwork.ts

```diff
@@ -205,41 +200,44 @@ export const useLocalZoneNetworks = (
   networks: TAggregatedNetwork[],
   pagination: PaginationState,
   filters: Filter[] = [],
-) =>
-  useQuery({
-    queryKey: getLocalZoneRegionsQueryKey(projectId),
-    queryFn: () =>
-      Promise.all(
-        networks.map(async (network) => {
-          const detailSubnet = await getSubnets(
-            projectId,
-            network.id,
-            network.region,
-          );
-
-          return {
-            ...network,
-            ...detailSubnet,
-            id: network.id,
-            name: network.name,
-            allocatedIp: detailSubnet.allocationPools
-              .map((ipPool) => `${ipPool.start} - ${ipPool.end}`)
-              .join(' ,'),
-          };
-        }),
-      ),
-    select: (result) => {
-      const mappedData = result.map((network) => ({
-        ...network,
-        search: `${network.name} ${network.region} ${network.dhcpEnabled} ${network.cidr} ${network.allocatedIp}`,
-      }));
-
-      return paginateResults(
-        applyFilters(mappedData || [], filters),
-        pagination,
-      );
-    },
+) => {
+  const subnetQueries = useQueries({
+    queries: networks?.map((n) => ({
+      queryKey: ['subnet', projectId, n.id, n.region],
+      queryFn: () => getSubnets(projectId, n.id, n.region),
+      select: (subnet: TSubnet) => {
+        const network = networks.find((net) => net.id === n.id);
+        const ips = subnet.allocationPools
+          ?.map((ipPool) => `${ipPool.start} - ${ipPool.end}`)
+          .join(' ,');
+        return {
+          ...subnet,
+          ...network,
+          allocatedIp: ips,
+          search: `${network.id} ${network.name} ${network.region} ${subnet.cidr} ${subnet.gatewayIp} ${ips}`,
+        };
+      },
+    })),
   });
+  const isPending = subnetQueries.some((q) => q.isPending);
+  const isFetching = subnetQueries.some((q) => q.isFetching);
+  const data = useMemo(() => {
+    if (isFetching || isPending) return undefined;
```

Comment:
You can't return a empty array as you are in useMemo or it's a specific case where you need to return undefined ?

Reply:
i prefer to return undefined because by default useQuery returns undefined also when data is not available instead of an empty array

Reply:
Ok for me